### ISSUE TITLE
API Gateway does not seem to be able to handle routing messages to my Lambda function when a body is present

### ISSUE THREAD

**Issue by rraina91** [localstack/localstack#1836]
Just converted my project from SAM over to Serverless and am attempting to get a good SDLC flow setup where I can test my API Gateway exposed lambda function. I have been able to configure it in a way where my "serverless deploy --local stage" is able to push it into my localstack environment properly and it presents the URL to call to me. I can use that in Postman to hit a basic "ping" endpoint I have setup which responds to a GET request and just returns the current time. This is working correctly and updates as expected every time I hit it (i.e. I am not looking at a stale cached result.)

However, when I tried to hit another endpoint with a POST body, the lambda_api/executors broke processing the input. In fact, if I add a POST body to the same GET message (illogical I know, but worth the test), it broke processing in the same way. I then picked a URL that I knew wasn't even valid, but with a POST and body and got the same breakage. I think something's just wrong with the body processing currently that perhaps my environment is encountering but I'm not exactly sure how to go about troubleshooting it at this point.

The message I get when I have a POST body of any sort is:

```
2019-12-06T20:25:49:WARNING:localstack.services.awslambda.lambda_api: Error executing Lambda function arn:aws:lambda:us-east-2:000000000000:function:tgs-local-handleRequest: Object of type bytes is not JSON serializable Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 359, in run_lambda
    event, context=context, version=version, asynchronous=asynchronous)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 82, in execute
    return do_execute()
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 67, in do_execute
    result, log_output = self._execute(func_arn, func_details, event, context, version)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 644, in _execute
    process.run()
  File "/usr/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 639, in do_execute
    result = lambda_function(event, context)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 484, in execute
    event, context, handler=handler, main_file=main_file)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 655, in execute_java_lambda
    save_file(event_file, json.dumps(event))
  File "/usr/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

I'm using a local executor, so I was able to narrow it down to what appears to be a problem breaking "save_file" here where it's attempting to serialize out the body into a file so it can be handed to my lambda and I guess it somehow is being encoded somewhere explicitly or implicitly as bytes and can't be.

 ```
def execute_java_lambda(self, event, context, handler, main_file):
        event_file = EVENT_FILE_PATTERN.replace('*', short_uid())
        save_file(event_file, json.dumps(event))
        TMP_FILES.append(event_file)
        class_name = handler.split('::')[0]
        classpath = '%s:%s:%s' % (LAMBDA_EXECUTOR_JAR, main_file, Util.get_java_classpath(main_file))
        cmd = 'java -cp %s %s %s %s' % (classpath, LAMBDA_EXECUTOR_CLASS, class_name, event_file)
        result, log_output = self.run_lambda_executor(cmd)
        LOG.debug('Lambda result / log output:\n%s\n> %s' % (
            result.strip(), log_output.strip().replace('\n', '\n> ')))
        return result, log_output
```

Any help or insight on how to better identify where the bad bytes are coming from would be appreciated.

**Comment by whummer** [comment]
Thanks for the detailed report @rraina91 . We can fix this issue by making the JSON dump resilient to bytes content, but would be interesting to understand the root cause of this issue.

Do you think you could post a small, self-contained sample project (serverless config + Java Lambda handler) that allows us to reproduce this issue? Thanks

**Comment by whummer** [comment]
Thanks for the update @gregghz @rraina91 . We believe this should be fixed in #2613 - can you please do a `docker pull localstack/localstack` and give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
